nilearn/maskers/nifti_labels_masker.py

```diff
@@ -325,82 +365,135 @@ def fit(self, imgs=None, y=None):
-                logger.log("resampling the mask", verbose=self.verbose)
+                            labels_img=self.labels_img,
+                        )
+                    )
+
+                if not np.allclose(
+                    self.mask_img_.affine,
+                    self.labels_img_.affine,
+                ):
+                    raise ValueError(
+                        _utils._compose_err_msg(
+                            'Regions and mask do not have the same affine.',
+                            mask_img=self.mask_img,
+                            labels_img=self.labels_img,
+                        ),
+                    )
+
+            elif self.resampling_target == 'labels':
+                _utils.logger.log('resampling the mask', verbose=self.verbose)
                 self.mask_img_ = image.resample_img(
                     self.mask_img_,
                     target_affine=self.labels_img_.affine,
                     target_shape=self.labels_img_.shape[:3],
-                    interpolation="nearest",
+                    interpolation='nearest',
                     copy=True)
+
             else:
                 raise ValueError(
-                    "Invalid value for "
-                    f"resampling_target: {self.resampling_target}"
+                    'Invalid value for '
+                    f'resampling_target: {self.resampling_target}'
                 )
 
             # Just check that the mask is valid
             masking._load_mask_img(self.mask_img_)
 
-        if not hasattr(self,
-                       '_resampled_labels_img_'):
+        if not hasattr(self, '_resampled_labels_img_'):
             # obviates need to run .transform() before .inverse_transform()
             self._resampled_labels_img_ = self.labels_img_
 
         if self.reports:
             self._reporting_data = {
                 'labels_image': self._resampled_labels_img_,
                 'mask': self.mask_img_,
-                'img': imgs}
+                'img': imgs,
+            }
         else:
             self._reporting_data = None
 
+        # Infer the number of elements in the mask
+        # This is equal to the number of unique values in the label image,
+        # minus the background value.
+        self.n_elements_ = np.unique(
+            image.get_data(self._resampled_labels_img_)
+        ).size - 1
+
         return self
 
     def fit_transform(self, imgs, confounds=None, sample_mask=None):
-        """ Prepare and perform signal extraction from regions.
+        """Prepare and perform signal extraction from regions.
+
+        Parameters
+        ----------
+        imgs : 3D/4D Niimg-like object
+            See http://nilearn.github.io/manipulating_images/input_output.html
+            Images to process. It must boil down to a 4D image with scans
+            number as last dimension.
+
+        confounds : CSV file or array-like or :obj:`pandas.DataFrame`, optional
```

Comment:
I think the one in `multi_nifti_masker` would be a list of objects, with the entries being equivalent to the parameter here.

Reply:
I don't know about making `array-like` more specific. I think `array-like` means that it could be a numpy array, a list of lists, or possibly some other objects with array-like organization. I don't know what specific checks/conversions are done, though, so I am not sure about specifically using `numpy

Reply:
Ok fine to leave as is then!